### PR TITLE
Add chronological sorting

### PR DESCRIPTION
**Pull request by adlez27**
Basic sort options have been reorganized into a dropdown menu with the following options
- Sort by sample, then by start time (new default)
- Sort by sample, then by entry name
- Sort by entry name only

image

Chronological sorting improves the experience of navigating through the entry list, as the graph will gradually scroll left to right.

sort chronological

### CODE REVIEW

File: resources/common/plugins/macro/sort-entries/plugin.json [hunk 1]
```diff
@@ -50,14 +50,36 @@
                 "defaultValue": true
             },
             {
-                "type": "boolean",
-                "name": "prioritizeEntryName",
+                "type":"enum",
```
Comment (sdercolin) [inline resources/common/plugins/macro/sort-entries/plugin.json:53]: Could you apply a formatter to it?

File: resources/common/plugins/macro/sort-entries/plugin.json [hunk 2]
```diff
@@ -10,9 +10,9 @@
     "author": "sdercolin",
     "email": "sder.colin@gmail.com",
     "description": {
-        "en": "Reorder all entries by entry name, sample name or tag",
+        "en": "Reorder all entries by entry name, sample name, start time, or tag",
         "zh": "根据条目名，采样名或标签来重新排序所有条目",
```
Comment (sdercolin) [inline resources/common/plugins/macro/sort-entries/plugin.json:14]:
```suggestion
        "zh": "根据条目名，采样名，开始时间或标签来重新排序所有条目",
```

File: resources/common/plugins/macro/sort-entries/plugin.json [hunk 3]
```diff
@@ -50,14 +50,36 @@
...
-                "name": "prioritizeEntryName",
+                "type":"enum",
+                "name":"priority",
                 "label": {
-                    "en": "Prioritize entry name over sample name",
-                    "zh": "根据条目名 > 采样名的顺序排序",
-                    "ja": "エントリ名 > サンプル名の順で並び替える"
+                    "en":"Sort entries by",
+                    "zh": "",
+                    "ja": ""
```
Comment (sdercolin) [inline resources/common/plugins/macro/sort-entries/plugin.json:58]:
```suggestion
                    "zh": "排序方式",
                    "ja": "並び替えの方法"
```

File: resources/common/plugins/macro/sort-entries/plugin.json [hunk 4]
```diff
@@ -50,14 +50,36 @@
...
+                    },
+                    {
+                        "en": "Sample -> Entry name",
+                        "zh": "",
+                        "jp": "サンプル名とエントリ名で並び変える"
+                    },
+                    {
+                        "en": "Entry name only",
+                        "zh": "",
+                        "jp": "エントリ名のみで並び替える"
```
Comment (sdercolin) [inline resources/common/plugins/macro/sort-entries/plugin.json:80]:
```suggestion
                        "en": "Sample -> Start time",
                        "zh": "采样名 -> 开始时间",
                        "jp": "サンプル名 -> 開始位置"
                    },
                    {
                        "en": "Sample -> Entry name",
                        "zh": "采样名 -> 条目名",
                        "jp": "サンプル名 -> エントリ名"
                    },
                    {
                        "en": "Entry name only",
                        "zh": "仅条目名",
                        "jp": "エントリ名のみ"
```

**Review by sdercolin — COMMENTED** [review]
Basically LGTM 👍

File: resources/common/plugins/macro/sort-entries/plugin.json [hunk 5]
```diff
@@ -86,4 +86,4 @@
     "scriptFiles": [
         "sort-entries.js"
     ]
-}
+}
```
Comment (sdercolin) [inline resources/common/plugins/macro/sort-entries/plugin.json:89]: Sorry for the detailed thing, but could you retain the empty line at the end of file?

Comment (adlez27) [reply resources/common/plugins/macro/sort-entries/plugin.json]: Sure, looks like the formatter removed it without me noticing

**Pull request merged**